runtime/AK/Set.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, Mustafa Quraish <mustafa@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "AK/HashTable.h"
+#include <initializer_list>
+
+namespace AK {
+
+template<typename T, typename TraitsForT, bool IsOrdered>
+class Set : public HashTable<T, TraitsForT, IsOrdered> {
+private:
+    using HashTableType = HashTable<T, TraitsForT, IsOrdered>;
+
+public:
+    using HashTableType::HashTable;
+
+    Set(std::initializer_list<T> list)
+    {
+        ensure_capacity(list.size());
+        for (auto& item : list)
+            add(item);
+    }
+
+    using HashTableType::contains;
+
+    HashSetResult add(const T& value) { return HashTableType::set(value); }
+    HashSetResult add(T&& value) { return HashTableType::set(move(value)); }
+    ErrorOr<HashSetResult> try_add(const T& value) { return HashTableType::try_set(value); }
+    ErrorOr<HashSetResult> try_add(T&& value) { return HashTableType::try_set(move(value)); }
```

Comment:
For these we probably don't need infallible versions and can drop the `try_` prefix, that's mostly a artefact from moving towards OOM fallibility